recipes/mp-units/0.7.0/conanfile.py

```diff
@@ -1,70 +1,121 @@
-from conans import ConanFile, CMake, tools
-from conans.tools import Version, check_min_cppstd
-from conans.errors import ConanInvalidConfiguration
 import os
 
-required_conan_version = ">=1.33.0"
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, cmake_layout
+from conan.tools.files import copy, get, rmdir
+from conan.tools.scm import Version
+
+required_conan_version = ">=2.0.0"
+
 
 class MPUnitsConan(ConanFile):
     name = "mp-units"
     homepage = "https://github.com/mpusz/units"
-    description = "Physical Units library for C++"
-    topics = ("units", "dimensions", "quantities", "dimensional-analysis", "physical-quantities", "physical-units", "system-of-units", "cpp23", "cpp20", "library", "quantity-manipulation")
+    description = "Physical Quantities and Units library for C++"
+    topics = (
+        "units",
+        "dimensions",
+        "quantities",
+        "dimensional-analysis",
+        "physical-quantities",
+        "physical-units",
+        "system-of-units",
+        "system-of-quantities",
+        "isq",
+        "si",
+        "library",
+        "quantity-manipulation",
+    )
     license = "MIT"
     url = "https://github.com/conan-io/conan-center-index"
     settings = "os", "arch", "compiler", "build_type"
-    generators = "cmake_find_package_multi"
+    requires = ["gsl-lite/0.38.0", "fmt/7.1.3"]
     no_copy_source = True
+    generators = "CMakeToolchain", "CMakeDeps"
 
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _min_cppstd(self):
+        return "20"
 
-    def requirements(self):
+    @property
+    def _minimum_compilers_version(self):
+        return {"gcc": "10.3", "clang": "12", "apple-clang": "13", "msvc": "1928"}
+
+    @property
+    def _full_compiler_version(self):
         compiler = self.settings.compiler
-        self.requires("fmt/7.1.3")
-        self.requires("gsl-lite/0.38.0")
-        if compiler == "clang" and compiler.libcxx == "libc++":
+        if compiler == "msvc":
+            if compiler.update:
+                return int(f"{compiler.version}{compiler.update}")
+            else:
+                return int(f"{compiler.version}0")
+        else:
+            return compiler.version
+
+    @property
+    def _use_range_v3(self):
+        compiler = self.settings.compiler
+        version = Version(self.settings.compiler.version)
+        return "clang" in compiler and compiler.libcxx == "libc++" and version < 14
+
+    def requirements(self):
+        if self._use_range_v3:
             self.requires("range-v3/0.11.0")
```

Comment:
Is it possible to update these dependencies? Since `mp-units` is header-only, these should enable the `transitive_headers` option of dependencies.

```suggestion
    def requirements(self):
        self.requires("gsl-lite/0.40.0", transitive_headers=True)
        self.requires("fmt/9.1.0", transitive_headers=True)
        if self._use_range_v3:
            self.requires("range-v3/0.12.0", transitive_headers=True)
```

Reply:
I do not want to change the dependencies to the latest ones for this old release as it might break my user's code. I use the latest dependecies for the next realease that I am preparing right now.

I will set `transitive_headers` for `gsl-lite` and `range-v3`, but I am not sure about `fmt`. Should it provide `transitive_libs` rather then `transitive_headers`?

Reply:
Yeah, you can update those dependencies later in that case. It looks like you actually don't need to set the `transitive_` items if you specify the `package_type` as `header-library`, looking [here](https://github.com/conan-io/tribe/blob/main/design/026-requirements_traits.md#detailed-design). Apparently Conan will figure that part out for us automatically. Sorry about that. That should handle `fmt` properly depending on how it's configured, `shared`, `static`, or `header-only`.

Reply:
No worries, I just pushed a commit removing those traits.